lib/github.ml

```diff
@@ -175,42 +175,49 @@ let publish_doc ~dry_run ~msg:_ ~docdir ~yes p =
 let github_auth ~dry_run ~user token =
   Sos.read_file ~dry_run token >>= fun token -> Ok (strf "%s:%s" user token)
 
-let run_with_auth ~dry_run auth curl k =
-  let auth = strf "-u %s" auth in
-  Sos.run_io ~dry_run curl (OS.Cmd.in_string auth) k
+let run_with_auth ?(default_body = "") ~dry_run ~auth Curl.{ url; args } =
+  let args = "-u" :: auth :: args in
```

Comment:
What would you think about extracting a `val with_auth : auth: string -> t -> t` in `Curl` and test it?

Reply:
Good idea, done.